### ISSUE TITLE
Unstable JSX formatting

### ISSUE THREAD

**Issue by karl** [prettier/prettier#1703]
@vjeux found a case where JSX formatting is not stable (https://github.com/prettier/prettier/pull/1671#issuecomment-303595598).

```
<div className="yourScore">
  Your score: <span className="score">{`${mini.crosstable.users[sessionUserId]} - ${mini.crosstable.users[user.id]}`}</span>
</div>
```
```
<div className="yourScore">
  Your score:
  {" "}
  <span className="score">{`${mini.crosstable.users[sessionUserId]} - ${mini
    .crosstable.users[user.id]}`}</span>
</div>;
```

```
<div className="yourScore">
  Your score:
  {" "}
  <span className="score">{`${mini.crosstable.users[sessionUserId]} - ${mini.crosstable.users[user.id]}`}</span>
</div>;
```

**Comment by piotrf** [comment]
I have the same issue where:

```
        <ul>
          <li><code>children</code> — meant to make space for <code>StatusSymbol</code> (required)</li>
        </ul>
``` 

is being turned into this:

```
        <ul>
          <li>
            <code>children</code>
            {' '}
            —&nbsp;meant to make space for
            {' '}
            <code>StatusSymbol</code>
            {' '}
            (required)
          </li>
        </ul>
```

**Comment by karl** [comment]
@piotrf That formatting should be fixed up as part of #1120. With this your example code would be formatted as:

```
<ul>
  <li>
    <code>children</code> — meant to make space for <code>StatusSymbol</code>
    {" "}
    (required)
  </li>
</ul>
```

The next release of Prettier will have this fix.

**Comment by karl** [comment]
☝🏻 I have a PR that should fix the unstable formatting issue.
